sphinx/tutorial/rose/furthertopics/polling.rst

```diff
@@ -0,0 +1,156 @@
+* A ``send_letter`` task which is run once the letter is composed.
+* A ``bob`` task which we will be using to poll with.
+* A ``read_letter`` task which will run once the polling task is complete.
+
+It will need some runtime. Add the following to your ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [runtime]
+       [[root]]
+           script = sleep 10
+       [[compose_letter]]
+           script = sleep 5; echo 'writing a letter to Bob...'
+       [[send_letter]]
+           env-script = eval `rose task-env`
+           script = """
+                    sleep 5
+                    echo 'Hello Bob' > $ROSE_DATA/letter.txt
+                    sleep 10
+                    """
+
+       [[bob]]
+           script = rose task-run
+       [[read_letter]]
+           env-script = eval `rose task-env`
+           script = sleep 5; cat $ROSE_DATA/letter.txt
+           post-script = rm $ROSE_DATA/letter.txt
+
+
+Adding Polling
+--------------
+
+In the suite directory create an ``app`` directory.
+
+In the ``app`` directory create a directory called ``bob``.
+
+In the newly created ``bob`` directory, create a ``rose-app.conf`` file.
+
+Edit the ``rose-app.conf`` file to look like this:
+
+.. code-block:: rose
+
+   [poll]
+   delays=10*PT5S
+   test=test -e $ROSE_DATA/letter.txt
+
+   [command]
+   default=echo 'Ooh, a letter!'
+
+We now have an app that does the following:
+
+* Has a polling ``test`` that checks for the existence of a file.
+* Polls up to 10 times with 5 second delays between each attempt.
+* Prints a message once the polling test succeeds.
+
+.. note::
+
+   The ordering of the ``[poll]`` and ``[command]`` sections is not important.
+   In practice, it may be preferable to have the ``[command]`` section at
+   the top as that should contain the main command(s) being run by the app.
+
+Save your changes and run the suite using ``rose suite-run``.
+
+The suite should now run.
+
+Notice that ``bob`` finishes and triggers ``read_letter`` before
+``send_letter`` has completed. This is because the polling condition has
+been met, allowing the main command in ``bob`` to be run.
+
+
+Improving The Polling
+---------------------
+
+At present we have specified our own routine for testing for the existence
+of a particular file using the ``test`` option. However, rose provides a
+simpler method for doing this.
+
+Edit the ``rose-app.conf`` in your ``bob`` app to look like the following:
+
+.. code-block:: rose
+
+   [poll]
+   delays=10*PT5S
+   all-files=$ROSE_DATA/letter.txt
+
+   [command]
+   default=echo 'Ooh, a letter!'
+
+Polling is now making use of the ``all-files`` option, which allows you to
+specify a list of files to check the existence of. Save your changes and
+run the suite to confirm it still works.
+
+
+Available Polling Types
+-----------------------
+
+test and all-files are just two of the available polling options:
+
+``all-files``
+   A list of space delimited list of files which only passes if all file
```

Comment:
Typo (and missing reference to it being a test etc): amend to 'A **test consisting of a** ~~list of~~ space-delimited list of files ...' (or similar). Note the hyphen as it should strictly be 'space-delimited'. Exactly the same should be done for line 134. Also consider italicising 'all' (perhaps with 'only' too) and 'any' to highlight the difference between these two options.